src/panels/lovelace/cards/hui-shopping-list-card.ts

```diff
@@ -118,92 +118,82 @@ class HuiShoppingListCard extends LitElement implements LovelaceCard {
+            .title=${this.hass!.localize(
               "ui.panel.lovelace.cards.shopping-list.add_item"
-            )}"
+            )}
+            @click=${this._addItem}
           >
           </ha-icon>
-          <paper-item-body>
-            <paper-input
-              no-label-float
-              class="addBox"
-              placeholder="${this.hass!.localize(
-                "ui.panel.lovelace.cards.shopping-list.add_item"
-              )}"
-              @keydown="${this._addKeyPress}"
-            ></paper-input>
-          </paper-item-body>
+          <paper-input
+            no-label-float
+            class="addBox"
+            placeholder=${this.hass!.localize(
+              "ui.panel.lovelace.cards.shopping-list.add_item"
+            )}
+            @keydown=${this._addKeyPress}
+          ></paper-input>
         </div>
         ${repeat(
           this._uncheckedItems!,
           (item) => item.id,
-          (item, index) =>
+          (item) =>
             html`
               <div class="editRow">
                 <paper-checkbox
-                  slot="item-icon"
-                  id="${index}"
-                  ?checked="${item.complete}"
-                  .itemId="${item.id}"
-                  @click="${this._completeItem}"
                   tabindex="0"
+                  ?checked=${item.complete}
+                  .itemId=${item.id}
+                  @click=${this._completeItem}
                 ></paper-checkbox>
-                <paper-item-body>
-                  <paper-input
-                    no-label-float
-                    .value="${item.name}"
-                    .itemId="${item.id}"
-                    @change="${this._saveEdit}"
-                  ></paper-input>
-                </paper-item-body>
+                <paper-input
+                  no-label-float
+                  .value=${item.name}
+                  .itemId=${item.id}
+                  @change=${this._saveEdit}
+                ></paper-input>
               </div>
             `
         )}
         ${this._checkedItems!.length > 0
           ? html`
               <div class="divider"></div>
               <div class="checked">
-                <span class="label">
+                <span>
                   ${this.hass!.localize(
                     "ui.panel.lovelace.cards.shopping-list.checked_items"
                   )}
                 </span>
                 <ha-icon
                   class="clearall"
-                  @action=${this._clearItems}
-                  .actionHandler=${actionHandler()}
                   tabindex="0"
                   icon="hass:notification-clear-all"
-                  .title="${this.hass!.localize(
+                  .title=${this.hass!.localize(
                     "ui.panel.lovelace.cards.shopping-list.clear_items"
-                  )}"
+                  )}
+                  .actionHandler=${actionHandler()}
+                  @action=${this._clearItems}
```

Comment:
Why do we use `actionHandler` here instead of `click`?

Reply:
/Shrug